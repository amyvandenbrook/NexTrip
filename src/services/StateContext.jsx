import React, { createContext, useReducer } from 'react'
import Reducer from './Reducer.jsx'

const StateContext = createContext()

const StateProvider = ({ children, initialState }) => {
    const defaultState = {
        routeData: [],
        route: undefined,
        routeChanged: false,
        directionData: [],
        direction: '',
        stopData: [],
    }
    const [state, dispatch] = useReducer(Reducer, {
        ...defaultState,
        ...initialState,
    })

    return (
        <StateContext.Provider
            value={{
                ...state,
                dispatch,
            }}
        >
            {children}
        </StateContext.Provider>
    )
}

export { StateContext, StateProvider }
