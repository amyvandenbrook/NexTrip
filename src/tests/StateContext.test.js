import React, { useContext } from 'react'
import { act, create } from 'react-test-renderer'
import { StateContext, StateProvider } from '../services/StateContext.jsx'

describe('StateContext', () => {
    let state
    let component
    test('returns the state values and children', () => {
        component = create(
            <StateProvider>
                <h1>Mock child</h1>
            </StateProvider>
        )
        state = component.toJSON()
        expect(state).toMatchSnapshot()
    })
})
