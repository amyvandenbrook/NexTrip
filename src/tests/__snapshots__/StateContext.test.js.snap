// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateContext returns the state values and children 1`] = `
<h1>
  Mock child
</h1>
`;
